src/client/components/network-editor/network-analyser.js

```diff
@@ -0,0 +1,247 @@
+
+// TODO add list types, add boolean?
+export const ATTR_TYPE = {
+  STRING: 'STRING',
+  NUMBER: 'NUMBER',
+  UNKNOWN: 'UNKNOWN',  // attribute has a blank or unsupported value, like null
+  MISSING: 'MISSING'   // the node/edge does not have the attribute at all
+};
+
+const hiddenAttrs = {
+  node: new Set([ 'id' ]),
```

Comment:
Possibly also `parent` if we end up supporting compounds.  It's probably not worth addressing for now, but it may be worthwhile to mark this line with a TODO so it's easy to address later.